File: src/main/java/com/plugin/frege/stubs/types/FregeClassDeclElementType.java
package com.plugin.frege.stubs.types;

import com.plugin.frege.psi.FregePsiClass;
import com.plugin.frege.psi.impl.FregeClassDeclImpl;
import com.plugin.frege.stubs.FregeClassStub;
import org.jetbrains.annotations.NotNull;

public class FregeClassDeclElementType extends FregeClassElementType {
    public FregeClassDeclElementType(@NotNull String debugName) {
        super(debugName);
    }

    @Override
    public @NotNull String getExternalId() {
        return super.getExternalId() + ".CLASS_DECL";
    }

    @Override
    public FregePsiClass createPsi(@NotNull FregeClassStub stub) {
        return new FregeClassDeclImpl(stub, this);
    }
}

File: src/main/kotlin/com/plugin/frege/stubs/types/FregeClassDeclElementType.kt
package com.plugin.frege.stubs.types

import com.plugin.frege.psi.FregePsiClass
import com.plugin.frege.psi.impl.FregeClassDeclImpl
import com.plugin.frege.stubs.FregeClassStub

class FregeClassDeclElementType(debugName: String) : FregeClassElementType(debugName) {
    override fun createPsi(stub: FregeClassStub): FregePsiClass = FregeClassDeclImpl(stub, this)

    override fun getExternalId(): String = super.getExternalId() + ".CLASS_DECL"
}
